src/domain/admin/hub/HubTemplatesAdminPage.tsx

```diff
@@ -55,8 +56,19 @@ const HubTemplatesAdminPage: FC<HubTemplatesAdminPageProps> = ({
     lifecycleTemplates,
     id: templatesSetID,
   } = hubTemplatesData?.hub.templates ?? {};
-  const canvasCallout = getCanvasCallout(hubCanvasesData?.hub.collaboration?.callouts);
-  const canvases = canvasCallout?.canvases;
+
+  // assuming we'll provide templates for the canvases only from hub callout canvases
+  const canvases = getAllCanvasesOnCallouts(hubCanvasesData?.hub.collaboration?.callouts);
+  const findParentCalloutId = useCallback(
```

Comment:
Does this make sense, making `canvasId` of type `string | undefined` just to return `undefined` if `canvasId` is `undefined`?